server/src/main/java/org/apache/druid/metadata/SQLMetadataRuleManager.java

```diff
@@ -362,6 +325,12 @@ public List<Rule> getRulesWithDefault(final String dataSource)
   @Override
   public boolean overrideRule(final String dataSource, final List<Rule> newRules, final AuditInfo auditInfo)
   {
+    if (newRules == null) {
+      throw new IAE("Rules cannot be null.");
+    } else if (newRules.isEmpty() && config.getDefaultRule().equals(dataSource)) {
```

Comment:
Could you please explain this condition: `config.getDefaultRule().equals(dataSource)`?

Reply:
The `defaultRule` corresponds to the cluster level default. I will add a comment here explaining it.

Reply:
Oh wait, I think the IAE thrown in the next line explains it.

Reply:
I see, so the defaultRule in the config corresponds to the datasource's name in the default load rules.

Reply:
Added a javadoc for `getDefaultRule`.